s3/backend.go

```diff
@@ -273,7 +273,7 @@ func (s *s3) HeadObject(bucketName, objectName string) (*gofakes3.Object, error)
 	// TODO: When we support metadata we need to add it here.
 	metadata := map[string]string{
 		"Content-Type":  mime.TypeByExtension(objectName),
-		"Last-Modified": time.Now().UTC().Format(http.TimeFormat), // TODO: update this when object has metadata
+		"Last-Modified": res.Object.LastModified(),
```

Comment:
I think there are more places in `backend.go` that can now use the `LastModified` of the object if you search for "LastModified:".

Also can you extend the tests in `s3_test.go` to check the returned `LastModified` time? I think all minio client methods return it as part of the response.